tests/zero_code_change/tensorflow2_integration_tests.py

```diff
@@ -45,28 +45,31 @@ def get_keras_data():
     return (x_train, y_train), (x_test, y_test)
 
 
-def test_keras_v2(script_mode: bool = False, eager_mode: bool = True):
+def helper_test_keras_v2(script_mode: bool = False, eager_mode: bool = True):
     """ Works as intended. """
     smd.del_hook()
-
+    tf.keras.backend.clear_session()
     if not eager_mode:
         tf.compat.v1.disable_eager_execution()
     with SagemakerSimulator() as sim:
         model = get_keras_model_v2()
         (x_train, y_train), (x_test, y_test) = get_keras_data()
 
-        model.compile(
-            loss="sparse_categorical_crossentropy",
-            optimizer=tf.keras.optimizers.RMSprop(),
-            metrics=["accuracy"],
-        )
+        opt = tf.keras.optimizers.RMSprop()
         if script_mode:
             hook = smd.KerasHook(out_dir=sim.out_dir, export_tensorboard=True)
+            opt = hook.wrap_optimizer(opt)
```

Comment:
Was the test working without wrap_optimizer() before?

Reply:
yes, it was.  vanilla TF 2.x doesn't call get_gradients() or add_Weights that are in wrap_optimizer. so its a basically a don't care. added it here to catch any possible changes around this in future versions of TF.